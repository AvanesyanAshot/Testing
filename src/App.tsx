import React, { useEffect, useState } from "react";
import "./App.css";

function App() {
  const [data, setData] = useState(null as any);
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 1000);
  }, []);

  return (
    <div>
      {data && <div>data</div>}
      <h1>Hello world</h1>
      <button>Добавить</button>
      <input type="text" placeholder="input..." />
    </div>
  );
}

export default App;
