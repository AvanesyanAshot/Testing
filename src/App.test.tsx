/* eslint-disable testing-library/no-debugging-utils */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App from "./App";

describe("App", () => {
  test("рендер Hello world", () => {
    render(<App />);
    const helloWorld = screen.getByText(/hello world/i);
    const btn = screen.getByRole("button");
    const input = screen.getByPlaceholderText(/input/i);

    expect(helloWorld).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
    screen.debug();
  });
  test("query input", () => {
    render(<App />);
    const btn = screen.queryByText(/test/i);
    expect(btn).toBeNull();
  });

  test("find input", async () => {
    render(<App />);
    screen.debug();
    const btn = await screen.findByText(/data/i);
    expect(btn).toBeInTheDocument();
    screen.debug();
  });
});
