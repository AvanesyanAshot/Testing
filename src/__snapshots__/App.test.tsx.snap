// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App рендер Hello world 1`] = `
<input
  placeholder="input..."
  type="text"
/>
`;

exports[`рендер Hello world 1`] = `
<input
  placeholder="input..."
  type="text"
/>
`;
